### ISSUE TITLE
Problem with Travis CI check

### ISSUE THREAD

**Issue by vojbarzz** [palantir/bulldozer#190]
Hello,

I'm not able to resolve issue if Travis CI checks are enabled for github repo. It consists of two checks (`continuous-integration/travis-ci/pr` and `continuous-integration/travis-ci/push`) and even if passed bulldozer thinks that it is not true and produce following message

```xxxx/policy-bot-test#2 is deemed not mergeable because of unfulfilled status checks: [continuous-integration/travis-ci]"```

The strange is that it complains on not complete name of check - looks like a parent group ?

![image](https://user-images.githubusercontent.com/16055634/84955460-99aec580-b0f7-11ea-999a-3dea71169457.png)

**Comment by bluekeyes** [comment]
That is strange, as Bulldozer should read the values for required contexts directly from the GitHub API. If possible, could you do the following?

1. Share a screenshot of the branch protection settings for this branch
2. Share the output of the [Get Branch Protection](https://developer.github.com/v3/repos/branches/#get-branch-protection) API for this branch? I'm interested in the `contexts` key of the `required_status_checks` object, so you can remove everything else.

Last, does Travis CI use the new-style Checks API? I'm curious if `pr` and `push` are two check runs that are part of the same check suite and GitHub allows users make the whole suite required.

**Comment by aeweidne** [comment]
Is it possible that the latest commit does not have those status because of something like `[skip ci]`?

**Comment by vojbarzz** [comment]
I have downloaded Your code and finded out that github reports `continuous-integration/travis-ci` as a required status - that is in match with what You can see in branch protection setup:
![image](https://user-images.githubusercontent.com/16055634/85904842-0b7ac380-b80a-11ea-8ca9-69f3de8c8aa9.png)

But github returns `continuous-integration/travis-ci/pr` or/and `continuous-integration/travis-ci/push` as successful/failed/pending status. This is based on if PR/push builds are enabled, as You can see here
![image](https://user-images.githubusercontent.com/16055634/85904890-2e0cdc80-b80a-11ea-96f6-dac51bf3af30.png)

I'm not sure if it is a specific to our enterprise installation (we have enterprise version of github & travis) or also to a public travis and github too. I have fixed it by an ugly piece of code (I care just about PR checks):

```go
	// hack to handle travis statuses
	logger.Debug().Msgf("received required statuses: %v", requiredStatuses)
	for i, status := range requiredStatuses {
		if status == "continuous-integration/travis-ci" {
			requiredStatuses[i] = "continuous-integration/travis-ci/pr"
			break
		}
	}
```
It works fine for my usecase for over 1 week - but it is hardcoded :-(. I'm fine with this solution in my case. 

We use github 2.19. I'm not able to get a travis version - as it is enterprise edition it is possible that not the latest one.

**Comment by andreyzher** [comment]
Hi,

We experienced this as well. From speaking to the administrators of our Travis Enterprise instance, this is a limitation in the enterprise version.
Travis made the change on the public instance a [while back](https://blog.travis-ci.com/2018-09-27-deprecating-github-commit-status-api-for-github-apps-managed-repositories), but the change never made it into the enterprise version it seems.

I've been running with a similar workaround for about 5-6 weeks now without any issues, but the code isn't pretty. Might be worth adding to a known issues/limitation section?

**Comment by bluekeyes** [comment]
I dug into this a bit more and it's weirder than I expected. GitHub (at least as of GHE 2.20, where I tested) seems to implement special behavior for the `continuous-integration/travis-ci` check and I was able to replicate the behavior here by posting statuses with `curl`.

1. Posting any/all of:

   - `continuous-integration/travis-ci`
   - `continuous-integration/travis-ci/push`
   - `continuous-integration/travis-ci/pr`
   
   results in _only_ `continuous-integration/travis-ci` appearing as an option in the protected branch status selector.
2. Making `continuous-integration/travis-ci` a required status causes `continuous-integration/travis-ci/pr` and `continuous-integration/travis-ci/push` to inherit the required flag.
3. This only works for `push` and `pr`. A status like `continuous-integration/travis-ci/branch` behaves normally.
4. This only works for `travis-ci`. A status like `continuous-integration/fake-ci/pr` behaves normally.

I can only assume that GitHub implemented this due to the widespread use of Travis CI at the time protected branches were introduced, but it's really surprising and I haven't found any documentation about it yet.

It sounds like the "correct" solution is to also make Bulldozer aware of this special logic implemented in GitHub.
